### PR TITLE
upgrade postgres versions to the latest ones, partial fix for #43

### LINKED ISSUE #43: releasing newer versions
Hi,

Do you have plans to release newer versions of the postgres binaries (10.19, 11.14, 12.9, 13.5, 14.1)? Especially 14.1 would be great as this contains 'fat' binaries for MacOS (IAW they run natively on the M1). 

-h

### PR DESCRIPTION
**Pull request by t0r0X**
New versions of PostgreSQL were released, with security and bug fixes:
https://www.postgresql.org/about/news/postgresql-141-135-129-1114-1019-and-9624-released-2349/
This partially fixes #43.

### CODE REVIEW

**Comment by t0r0X** [comment]
The checks seem to fail because of a expired certificate from `nixos.org`:
```
Processing triggers for libc-bin (2.19-0ubuntu6.15) ...
+ wget -O patchelf.tar.gz https://nixos.org/releases/patchelf/patchelf-0.9/patchelf-0.9.tar.gz
--2021-11-25 10:12:53--  https://nixos.org/releases/patchelf/patchelf-0.9/patchelf-0.9.tar.gz
Resolving nixos.org (nixos.org)... 54.205.240.192, 67.207.80.24, 2600:1f18:2489:8202:1542:e5:19c0:d28b, ...
Connecting to nixos.org (nixos.org)|54.205.240.192|:443... connected.
ERROR: cannot verify nixos.org's certificate, issued by '/C=US/O=Let\'s Encrypt/CN=R3':
  Issued certificate has expired.
To connect to nixos.org insecurely, use `--no-check-certificate'.
+ cp '/usr/lib/*/libssl.so.1.0.0' '/usr/lib/*/libcrypto.so.1.0.0' ./lib
cp: cannot stat '/usr/lib/*/libssl.so.1.0.0': No such file or directory
cp: cannot stat '/usr/lib/*/libcrypto.so.1.0.0': No such file or directory
```

I looked at that site's certificate, and it seems they have refreshed it, please could someone restart the checks? Or, what's the recommended procedure for these cases?

**Comment by tomix26** [comment]
I guess the docker image that is used to build the project does not contain the new certificate. That's because the build pipeline rely on older images with older operating systems for backwards compatibility of the resulting binaries. So it will be necessary to either add the `--no-check-certificate` option to each call of the wget and curl commands, or to incorporate the new certificate to the older images. The first option is a potential security threat, so the second option is preferred. But don't worry I'll look into this during the weekend and try to sort it out.

**Comment by t0r0X** [comment]
Oh, now I understand the issue. I'd also prefer the second option (add new certificate to older images), better safe than sorry. The concerned image is `ubuntu:14.04`. 

I'm not worried :-) just trying to understand, learn and help, if possible.

**Comment by t0r0X** [comment]
Beside the `ubuntu:14.04` CA issue, there is another issue for `PostgreSQL 13.5` on MacOS, in `scripts/test-postgres-darwin.sh` line 40:
```
++ psql -qAtX -h localhost -p 65432 -U postgres -d postgres -c 'SET jit_above_cost = 10; SELECT SUM(relpages) FROM pg_class;'
ERROR:  could not load library "/private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tmp.ewEBRd46/pg-test/lib/postgresql/llvmjit.so": dlopen(/private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tmp.ewEBRd46/pg-test/lib/postgresql/llvmjit.so, 10): Library not loaded: @loader_path/../..//opt/local/lib/libncurses.6.dylib
2021-11-25 10:50:23.236 UTC [5220] ERROR:  could not load library "/private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tmp.ewEBRd46/pg-test/lib/postgresql/llvmjit.so": dlopen(/private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tmp.ewEBRd46/pg-test/lib/postgresql/llvmjit.so, 10): Library not loaded: @loader_path/../..//opt/local/lib/libncurses.6.dylib
  Referenced from: /private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tmp.ewEBRd46/pg-test/lib/postgresql/llvmjit.so
  Reason: no suitable image found.  Did find:
	file system relative paths not allowed in hardened programs
```

**Comment by t0r0X** [comment]
Sorry for the commit spam, it took me a while to get used to the Github actions and to understand how to run them from a feature branch.

File: .github/workflows/gradle.yml [hunk 1]
```diff
@@ -1,12 +1,13 @@
 name: Checks
-on: [push, pull_request]
+on: [ push, pull_request ]
 jobs:
   linux:
     name: PostgreSQL ${{ matrix.postgres }} (Linux)
     runs-on: ubuntu-latest
     strategy:
+      fail-fast: false
```
Comment (tomix26) [inline .github/workflows/gradle.yml:8]: Very useful, I didn't know it's possible to configure it in this way.

File: scripts/test-postgres-darwin.sh [hunk 2]
```diff
@@ -32,13 +32,21 @@ tar -xJf $TRG_DIR/pg-dist/$ZIP_FILE -C $TRG_DIR/pg-test
...
 
+# Shutdown DB server and do cleanup on exit
+function cleanup() {
+  local errcode=$?
+  $TRG_DIR/pg-test/bin/pg_ctl -w -D $TRG_DIR/pg-test/data stop
+  rm -rf $TRG_DIR
+  return $errcode
+}
+
+trap cleanup EXIT
```
Comment (tomix26) [inline scripts/test-postgres-darwin.sh:43]: Also a very smart solution 👍

**Review by tomix26 — COMMENTED** [review]
I'm going to fix the missing library in macos build and then we can merge it. Good job!

**Review by tomix26 — APPROVED** [review]

**Comment by tomix26** [comment]
@t0r0X Thank you very much for your contribution. I really appreciate it. It saved me a lot of time.

File: .github/workflows/gradle.yml (hunk 1 above)
Comment (t0r0X) [reply .github/workflows/gradle.yml:8]: It was really coincidence that I found that setting.

File: scripts/test-postgres-darwin.sh (hunk 2 above)
Comment (t0r0X) [reply scripts/test-postgres-darwin.sh:43]: Thanks. Those 6 hours timeout bugged me until I found the cause :-)

**Comment by t0r0X** [comment]
@tomix26 I'm glad I could help. I was searching for a way to replace H2 with PostgreSQL, for my local development and test when I stumbled upon the `embedded-postgres`/`embedded-database` projects. Great stuff, thanks! And I learned a lot!

**Pull request merged**